### PR TITLE
Add option to hide games with mods

### LINKED ISSUE #641: option to hide modded games
in addition to the option to hide private games

### PR DESCRIPTION
**Pull request by 20seven**
Added a checkbox to give user the option to hide modded games
Fixes #641

### CODE REVIEW

File: src/main/resources/i18n/messages_de_DE.properties [hunk 1]
```diff
@@ -95,6 +95,7 @@ games.create=Er
 games.create.minRating=Min.
 games.create.maxRating=Max.
 games.showPasswordProtectedGames=Zeige passwortgesch�tzte Spiele
+games.showModdedGames=Zeige modifizierte Spiele
```
Comment (1-alex98) [inline src/main/resources/i18n/messages_de_DE.properties]: Bad translation

File: src/main/java/com/faforever/client/game/CustomGamesController.java [hunk 2]
```diff
@@ -122,15 +123,32 @@ public TilesSortingOrder fromString(String string) {
...
   }
 
-  public void onShowPrivateGames(ActionEvent actionEvent) {
-    CheckBox checkBox = (CheckBox) actionEvent.getSource();
-    boolean selected = checkBox.isSelected();
-    if (selected) {
-      filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE);
-    } else {
-      filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE.and(gameInfoBean -> !gameInfoBean.getPasswordProtected()));
+    public void onShowPrivateGames(ActionEvent actionEvent) {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/CustomGamesController.java]: You should save the state of the Checkbox in the preferences. Cosider binding the checkedProperty of the CheckBox to the property in the preferences. See https://github.com/FAForever/downlords-faf-client/blob/1bee186d0a576fd1dc00cdc10874bc4599f666db/src/main/java/com/faforever/client/preferences/ui/SettingsController.java#L168

File: src/main/resources/i18n/messages.properties [hunk 3]
```diff
@@ -158,6 +158,7 @@ games.create=Create game
 games.create.minRating=Min.
 games.create.maxRating=Max.
 games.showPasswordProtectedGames=Show password protected games
+games.showModdedGames=Show Modded Games
```
Comment (1-alex98) [inline src/main/resources/i18n/messages.properties]: Rather "Games with mods"

File: src/main/resources/i18n/messages_de_DE.properties (hunk 1 above)
Comment (1-alex98) [reply src/main/resources/i18n/messages_de_DE.properties]: Maybe "Zeige Spiele mit Mods" ;)

**Comment by axel1200** [comment]
Maybe write a small/quick unit test

**Comment by axel1200** [comment]
I just had a cool idea bind the visibility property(maybe not the visibilty property I dont know which property you must bind, but i am sure you will find out) of the password/mod column to the selected property of the check boxes or better to the preference properties... ;)

**Comment by 1-alex98** [comment]
@micheljung your turn

File: src/main/java/com/faforever/client/game/CustomGamesController.java [hunk 4]
```diff
@@ -96,14 +110,22 @@ public TilesSortingOrder fromString(String string) {
...
     filteredItems = new FilteredList<>(games);
-    filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE);
+    showModdedGamesCheckBox.selectedProperty().bindBidirectional(preferencesService.getPreferences().showModdedGamesProperty());
+    showPasswordProtectedGamesCheckBox.selectedProperty().bindBidirectional(preferencesService.getPreferences().showPasswordProtectedGamesProperty());
+
+    updateFilteredItems();
+    preferencesService.getPreferences().showModdedGamesProperty().addListener(new WeakChangeListener<>(filterConditionsChangedListener));
+    preferencesService.getPreferences().showPasswordProtectedGamesProperty().addListener(new WeakChangeListener<>(filterConditionsChangedListener));
+
+    preferencesService.getPreferences().showModdedGamesProperty().addListener(new WeakChangeListener<>(showModsColumnChangedListener));
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/game/CustomGamesController.java]: What about the password column?

Comment (1-alex98) [reply src/main/java/com/faforever/client/game/CustomGamesController.java]: Its small about we could also make that one disappear no? Did you test if that looks ok if they disappear and come back or if that messes up the column widths?

Comment (20seven) [reply src/main/java/com/faforever/client/game/CustomGamesController.java]: yea, I like it better with password column still there as it gives some margins to the border of the table. But latest commit removes pw column as well,. So if you want to check out how it looks, please do

**Comment by 1-alex98** [comment]
I like it ❤️

**Comment by 1-alex98** [comment]
image

**Comment by 1-alex98** [comment]
image

**Comment by 1-alex98** [comment]
@20seven Please resolve conflicts by rebasing on orign/develop

File: src/main/java/com/faforever/client/game/CustomGamesController.java [hunk 5]
```diff
@@ -80,6 +93,9 @@ public CustomGamesController(UiService uiService, GameService gameService, Prefe
   }
 
   public void initialize() {
+    gamesTableController = uiService.loadFxml("theme/play/games_table.fxml");
+    gamesTilesContainerController = uiService.loadFxml("theme/play/games_tiles_container.fxml");
```
Comment (micheljung) [inline src/main/java/com/faforever/client/game/CustomGamesController.java]: This is the only thing that bugs me about this PR; we're loading both controllers even though we're only displaying one.
The client already uses more memory than I want it to, so if we can find a better solution I'd appreciate that. I didn't take the time yet to take a closer look and if I do, I'd like to discuss it directly in Discord ;)

**Pull request merged**